collector/zfs_linux.go

```diff
@@ -27,31 +26,110 @@ import (
 )
 
 const (
-	zfsArcstatsProcpath = "spl/kstat/zfs/arcstats"
+	zfsProcpathBase      = "spl/kstat/zfs/"
+	zfsArcstatsExt       = "arcstats"
+	zfsDmuTxExt          = "dmu_tx"
+	zfsFmExt             = "fm"
+	zfsFetchstatsExt     = "zfetchstats"
+	zfsVdevCacheStatsExt = "vdev_cache_stats"
+	zfsXuioStatsExt      = "xuio_stats"
+	zfsZilExt            = "zil"
 )
 
-func (c *zfsCollector) openArcstatsFile() (file *os.File, err error) {
-	file, err = os.Open(procFilePath(zfsArcstatsProcpath))
+func (c *zfsCollector) openProcFile(path string) (file *os.File, err error) {
+	file, err = os.Open(procFilePath(path))
 	if err != nil {
-		log.Debugf("Cannot open %q for reading. Is the kernel module loaded?", procFilePath(zfsArcstatsProcpath))
+		log.Debugf("Cannot open %q for reading. Is the kernel module loaded?", procFilePath(path))
 		err = zfsNotAvailableError
 	}
 	return
 }
 
 func (c *zfsCollector) updateArcstats(ch chan<- prometheus.Metric) (err error) {
-	file, err := c.openArcstatsFile()
+	file, err := c.openProcFile(zfsProcpathBase + zfsArcstatsExt)
```

Comment:
Thanks again, hadn't realized there was a library API for it (but probably should have looked).